src/libostree/ostree-repo.c

```diff
@@ -517,6 +1042,10 @@ ostree_repo_finalize (GObject *object)
   g_clear_pointer (&self->remotes, g_hash_table_destroy);
   g_mutex_clear (&self->remotes_lock);
 
+  GHashTable *lock_table = g_private_get (&repo_lock_table);
+  if (lock_table)
+    g_hash_table_remove (lock_table, self);
```

Comment:
Maybe add:
```
if (g_hash_table_size (lock_table) == 0)
  g_private_replace (&repo_lock_table, NULL);
```
to save a bit of memory if `OstreeRepo` is only used sporadically throughout a process’ lifetime.

Reply:
Oh, that's nice. I think it's almost always the case that applications are operating on one repository, so keeping the hash table around all the time always bugged me.